### DISCUSSION TITLE
Log Failure Alert?

### DISCUSSION THREAD

**Discussion by DinhDiesel** [Security-Onion-Solutions/securityonion#6813] [Unsupported Versions]
I was wondering if Security Onion is capable of a log failure alert? Or is there way to alert if there is a failure to log an event? Or lets say if the syslog daemon service stops running is there a way to send alerts for that also?

**Comment by weslambert** [comment]
If you want to monitor lack of new events for a particular log type (or multiple), you could implement an Elastalert rule of the flatline type: 

https://elastalert.readthedocs.io/en/latest/elastalert.html
https://elastalert.readthedocs.io/en/latest/ruletypes.html#flatline

**Reply by DinhDiesel** [reply]
> @weslambert thank you for pointing me in the right direction! I've created a syslogflatline.yaml for the flatline rule under /opt/elastalert/rules 

es_host: es host ip
es_port: 9200
index: '*:elastalert_status-*'
name: Syslog Flatline
type: flatline
threshold: 1
query_key: log.source.address
timeframe:
minutes: 30
falsepositives:
None
fields:
- log.source.address
priority: 2
realert:
mintues: 0
alert:
- "email"
email:
- user@domain.com
- smtp_host: ip smtp host
smtp_port: 25
from_addr: elastalert@domain.com

I was unable to get this to load not sure if its because a format issue?

**Reply by DinhDiesel** [reply]
I have added this rule in my manager/search node, looks like I cant reach my smtp host even though I can ping it. Also noticed this "/usr/local/lib/python3.9/site-packages/staticconf/loader.py". I went to /user/local/lib/...here is where I noticed a python3.6 folder and do not see a 3.9 would that effect this rule?


name: "Syslog_Flatline"
type: flatline
index: *:elastalert_status-*

threshold: 1 #if below one, alert triggered!

timeframe:
  minutes: 5

filter:
- query:
    query_string:
      query: "event.dataset: syslog*"
use_count_query: true

doc_type: _doc

alert:
  - "email" 

smtp: internal relay ip address

port: 25

email:
  - "user@domain"
![Error101_smtphostunreachable](https://user-images.githubusercontent.com/87437722/151592778-970281e8-17a0-415d-b0e9-b1720eac98ce.JPG)

**Reply by dougburks** [reply]
> looks like I cant reach my smtp host even though I can ping it.

Have you tried manually connecting to port 25 of your mail server to see if it's being blocked by a firewall or some other network device?

**Reply by DinhDiesel** [reply]
its just an internal relay I use for Office 365.

**Reply by weslambert** [reply]
I don't think Doug's question above was answered -- has the connection here been determined to be successful?

**Comment by IzacFrank** [comment]
@DinhDiesel I have the same email server setup and in my working alerts, the config is like the one below

```
alert:
- email

# (required, email specific)
# a list of email addresses to send alerts to

email:
- "receiver@xxx.com"
smtp_host: "x.x.x.x"
smtp_port: 25
# smtp_ssl: true
# smtp_auth_file: '/opt/elastalert/rules/smtp_auth_file.txt'
from_addr: "noreply@yyyyy.com"

```

Please also see the config on #7051
It may guide you about the format. I also like to get alerts when there is a failure like "rules update" failure etc. Trying to figure that out.

I hope it helps.

**Reply by IzacFrank** [reply]
@DinhDiesel Any progress? If it is working, could you please share your config? 

Thanks.

**Reply by DinhDiesel** [reply]
I set my yaml file up just like yours but no alerts are getting sent to my email.

**Comment by DinhDiesel** [comment]
I've been busy I will update my config asap I really appreciate it will update you @IzacFrank
